app/database/itemStore.go

```diff
@@ -38,9 +38,31 @@ func (s *ItemStore) Insert(data *Item) error {
 	return s.db.insert(s.tableName(), data)
 }
 
+// All creates a composable query without filtering
+func (s *ItemStore) All() *DB {
+	return &DB{s.db.Table(s.tableName())}
+}
+
 // IsValidHierarchy gets an ordered set of item ids and returns whether they forms a valid item hierarchy path from a root
 func (s *ItemStore) IsValidHierarchy(ids []int64) (bool, error) {
-	return false, nil
+	l := len(ids) - 1
```

Comment:
in my opinion, using a variable for that makes the code more complex to understand. it not even easy to express what `l` represents